### PR TITLE
Filter system

### LINKED ISSUE #77: Add a way to filter by any variable
For example filter by drg weight with brushing. Might use lineup and might require us extracting the lineup filtering component into something we can mount

### PR DESCRIPTION
**Pull request by haihan-lin**
Closes #77

### CODE REVIEW

File: frontend/src/Components/Charts/HeatMap/SingleHeatRow.tsx [hunk 1]
```diff
@@ -42,8 +42,9 @@ const SingleHeatRow: FC<Props> = ({ dataPoint, valueScaleDomain, valueScaleRange
                     return (
                         <g>
                             <Tooltip
-                                title={<div className="charttooltip">{outputContent}</div>}
+                                title={<div key={dataPoint.aggregateAttribute + '-' + point} className="charttooltip">{outputContent}</div>}
```
Comment (JackWilb) [inline frontend/src/Components/Charts/HeatMap/SingleHeatRow.tsx:45]: Please use the `this is a ${variable}` syntax here and below

File: frontend/src/Components/LayoutGenerator.tsx [hunk 2]
```diff
@@ -145,10 +146,10 @@ const LayoutGenerator: FC = () => {
                 })}
             </Responsive>
         </div >,
-        <div id="filter-tab">
-            <div className={"lineup"} id={"lineup-wrapper"}>
-                {/* <LineUpWrapper hemoglobinDataSet={hemoData} /> */}
-            </div>
+        <div id="filter-tab" style={{ height: "90vh" }}>
+            <FilterBoard />
+            {/* <LineUpWrapper hemoglobinDataSet={hemoData} /> */}
```
Comment (JackWilb) [inline frontend/src/Components/LayoutGenerator.tsx:151]: Let's make an issue to remove lineup if that's our next step

File: frontend/src/App.tsx [hunk 3]
```diff
@@ -18,14 +18,16 @@ export const DataContext = createContext<SingleCasePoint[]>([])
...
     const [outputFilteredData, setOutputFilteredDAta] = useState<SingleCasePoint[]>([])
 
 
     useEffect(() => {
-        if (process.env.REACT_APP_REQUIRE_LOGIN !== "true") { cacheHemoData(); }
-        if (store.configStore.isLoggedIn && hemoData.length === 0) {
+        if (process.env.REACT_APP_REQUIRE_LOGIN !== "true") {
+            cacheHemoData();
+        }
+        else if (store.configStore.isLoggedIn && hemoData.length === 0) {
```
Comment (JackWilb) [inline frontend/src/App.tsx:30]: You could simplify this to:

```
if (process.env.REACT_APP_REQUIRE_LOGIN !== "true" || (store.configStore.isLoggedIn && hemoData.length === 0))
```

I'm not sure this is more readable, though. I'll leave it up to you

File: frontend/src/Components/Utilities/LeftToolBox/CurrentView.tsx [hunk 4]
```diff
@@ -69,31 +60,24 @@ const CurrentView: FC<Props> = ({ totalCaseNum }: Props) => {
...
-                                <Chip size="small" label="Emergent" clickable color={surgeryUrgencySelection[2] ? "primary" : undefined}
-                                    onClick={() => { store.configStore.changeSurgeryUrgencySelection([surgeryUrgencySelection[0], surgeryUrgencySelection[1], !surgeryUrgencySelection[2]]) }} /></Box>} />
+                        <ListItemText primary="Surgery Urgency" secondary={
+                            `${surgeryUrgencySelection[0] ? "Urgent, " : ""}
+                            ${surgeryUrgencySelection[1] ? "Elective, " : ""}
+                            ${surgeryUrgencySelection[2] ? "Emergent" : ""}`} />
                     </ListItem>
+
+                    {/* Should we rename this to "use alternative color scheme or something like that?" */}
+
```
Comment (JackWilb) [inline frontend/src/Components/Utilities/LeftToolBox/CurrentView.tsx:80]: Remove this code and make into an issue

**Review by JackWilb — COMMENTED** [review]
Looks good. I'm going to do some testing on the server this afternoon and then I'll approve. I left a couple of suggestions for you, but they shouldn't hold back the PR. Feel free to make the changes if they make sense to you

File: frontend/src/Components/LayoutGenerator.tsx (hunk 2 above)
Comment (haihan-lin) [reply frontend/src/Components/LayoutGenerator.tsx:151]: I'm actually fine with removing lineup completely in this PR.

I have a local version of the old src, and that has lineup set up in there. So we want to add it back in, I can refer to my local files. 

I'll remove it completely.

File: frontend/src/Components/Utilities/LeftToolBox/CurrentView.tsx (hunk 4 above)
Comment (haihan-lin) [reply frontend/src/Components/Utilities/LeftToolBox/CurrentView.tsx:80]: existing issue #163

**Comment by haihan-lin** [comment]
- [ ] Add a filter board for the app
- [ ] the filter affect the available cases -> question: should the filter affect the "aggregated cases", for example -/4700 to -/90 because the filter changed. Currently this happens but I feel like it is confusing since surgery selection doesn't affect the number.
- [ ] Reset for each filter section
- [ ] Reset button should be disabled when it is at default filter state.

**Comment by JackWilb** [comment]
Things for me to double check:
- [ ] lineup remnants are fully gone
- [ ] test that it compiles and works as intended
- [ ] are there any related issues that we can close with this PR

**Review by JackWilb — APPROVED** [review]
Looks good to me! I noted a couple issues in my comment above that might warrant some further investigation

**Comment by haihan-lin** [comment]
> I noticed issues loading an old state. Saving and loading new states seems fine
> Aggregated cases showed 4712/7. I'm not sure what that means

Pretty sure this is because the main interface doesn't "reload" until you click back to render it. But I'll take a closer look on next PR.

**Pull request merged**